### PR TITLE
Add API reference documentation for `ci-cd`

### LINKED ISSUE #113: Add API reference documentation for this repository
It would be nice with some API reference about how this repository works for contributing developers.

### PR DESCRIPTION
**Pull request by CasperWA**
Part 2 of closing #95.

To do:
- [x] Rebase on `main` after #109 has been merged.

Add API reference documentation for _this_ repository to the documentation at [SINTEF.github.io/ci-cd](https://SINTEF.github.io/ci-cd).

Note, this is part 2 of 3 of a new set of PRs that replace #99.

Closes #113

### CODE REVIEW

File: .github/workflows/_local_ci_cd_updated_main.yml [hunk 1]
```diff
@@ -16,8 +16,9 @@ jobs:
       default_repo_branch: main
       update_docs: true
       doc_extras: "[docs]"
-      update_python_api_ref: false
+      update_python_api_ref: true
```
Comment (francescalb) [inline .github/workflows/_local_ci_cd_updated_main.yml:19]: Does this have anything to do with the issue?

Comment (francescalb) [reply .github/workflows/_local_ci_cd_updated_main.yml:19]: Ok, it doesn't really. This is just adding documentation here. I don't understand why that is the same issue about allowing for more detailed control of major and minor version updates for dependencies, though.

**Review by francescalb — COMMENTED** [review]
How do I test that the documentation is properly generated?

What are the codecov warnings about newly added lines? They do not seem to be newly added in this PR, but is sounds like that from the codecov issues.

File: .github/workflows/_local_ci_cd_updated_main.yml (hunk 1 above)
Comment (CasperWA) [reply .github/workflows/_local_ci_cd_updated_main.yml:19]: It isn't. It's completely separate, really :) I just did this in the previous PR, hence I separated it out to create this separate PR.

**Comment by CasperWA** [comment]
> How do I test that the documentation is properly generated?

You could generate it locally? Or you could check the CI/CD workflow that builds the documentation and that it doesn't fail? So that would be [this run](https://github.com/SINTEF/ci-cd/actions/runs/4355597736/jobs/7612434362).

> What are the codecov warnings about newly added lines? They do not seem to be newly added in this PR, but is sounds like that from the codecov issues.

I think this is a timing issue, since these PRs were created simultaneously, but built upon each other. So while the changes codecov reference have now been merged into `main` from the previous PR, codecov still ran when these changes weren't merged, so it flagged these lines in its run. I'm sure if it runs again, for some reason (if I add more commits) they should go away. We'll see I suppose.

**Review by francescalb — APPROVED** [review]
Good.

**Pull request merged**